src/components/modal/__internal__/modal-manager.ts

```diff
@@ -0,0 +1,94 @@
+type SetTriggerRefocusFlag = (boolean: boolean) => void;
+
+type ModalList = {
+  modal: HTMLElement;
```

Comment:
Could this type be tightened to the below?

```suggestion
  modal: HTMLDivElement;
```

Reply:
It could be, but then we're restricting the use of `useModalManager` to only `div` elements. I'm not sure if there's any benefit of restricting that to `HTMLDivElement` but I might be wrong

Reply:
Yeah I think it's not a big issue so I'm happy to leave it as is